src/main/java/com/pinterest/secor/consumer/Consumer.java

```diff
@@ -143,14 +156,17 @@ public void run() {
                 }
 
                 long now = System.currentTimeMillis();
-                if (nMessages++ % checkMessagesPerSecond == 0 ||
+                if (mDeterministicUploadPolicyTracker != null ||
```

Comment:
Could this still be non-deterministic dependent on the time taken within this loop? Will we be segmenting on disk before the upload (i.e. only uploading the portion that we would have uploaded had we checked in time?)